### PR TITLE
Add integer framework

### PR DESCRIPTION
**Pull request by mborland**
Adds the integer parser to `from_chars`

### CODE REVIEW

File: include/boost/charconv/from_chars.hpp [hunk 1]
```diff
@@ -15,6 +16,11 @@ namespace boost { namespace charconv {
 struct from_chars_result
 {
     const char* ptr;
+
+    // Values:
+    // 0 = no error
+    // 1 = invalid_argument
+    // 2 = result_out_of_range
```
Comment (pdimov) [inline include/boost/charconv/from_chars.hpp]: Please use the standard values of EINVAL and ERANGE from `<cerrno>`, to match the errc values.

**Comment by mborland** [comment]
@pdimov Any other comments before I merge this? I am happy with where it sits, and added a test set to ensure that the results matched the STL implementation.

**Comment by pdimov** [comment]
Looks good.

BOOST_CHARCONV_ASSERT should use BOOST_ASSERT instead of plain assert.

It's possible to use `BOOST_TEST( x == y )` instead of `BOOST_TEST_EQ( x, y )` in the case where `x` isn't streamable; there's no need to drop to assert.

constexpr does mean we need to put the integral implementation into the header after all, but it's probably worth it.

**Comment by mborland** [comment]
Do you know the solution for [these](https://github.com/CPPAlliance/charconv/actions/runs/3932369587/jobs/6724844470#step:6:13) cmake errors? It looks like Boost.Assert is being installed but not found?

**Comment by pdimov** [comment]
You probably need to regenerate the CMakeLists.txt file to pick up the dependency on Assert. Use `boostdep --cmake charconv` (assuming that the library is in libs/charconv in the Boost tree.)

**Pull request merged**